skipcpio/skipcpio.c

```diff
@@ -23,16 +23,65 @@
 #define _GNU_SOURCE
 #endif
 
-#include <stdbool.h>
 #include <stdio.h>
 #include <stdlib.h>
-#include <unistd.h>
 #include <string.h>
 
 #define CPIO_END "TRAILER!!!"
-#define CPIO_ENDLEN (sizeof(CPIO_END)-1)
+#define CPIO_ENDLEN (sizeof(CPIO_END) - 1)
+
+#define CPIO_ALIGNMENT 4
+
+struct cpio_header {
+        char c_magic[6];
+        char c_ino[8];
+        char c_mode[8];
+        char c_uid[8];
+        char c_gid[8];
+        char c_nlink[8];
+        char c_mtime[8];
+        char c_filesize[8];
+        char c_devmajor[8];
+        char c_devminor[8];
+        char c_rdevmajor[8];
+        char c_rdevminor[8];
+        char c_namesize[8];
+        char c_check[8];
+};
+
+struct buf_struct {
+        struct cpio_header h;
+        char filename[CPIO_ENDLEN];
+};
+
+static struct buf_struct buf;
```

Comment:
nit: you could turn this into a union with `char b[4096]` and reuse it for copy-buffer.

Reply:
done